streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/DruidBeamFactoryImpl.java

```diff
@@ -352,10 +353,7 @@ public StreamlineTimestamper(String timestampField) {
 
         @Override
         public DateTime timestamp(Map<String, Object> theMap) {
-            if (PROCESSING_TIME.equalsIgnoreCase(timestampField))
-                return new DateTime(System.currentTimeMillis());
-
-            return new DateTime(theMap.get(timestampField));
+            return new DateTime(theMap.get(timestampField), DateTimeZone.UTC);
```

Comment:
May be good to throw an exception or LOG error if the timestampField is missing in the payload. `new Datetime(null, DateTimeZone.UTC)` creates an object with current time so the payload can be sent to druid with current time.

Reply:
added an error log. Pl check.